### PR TITLE
Add Camera portal

### PR DESCRIPTION
**Pull request by aospiridonov**
Add Camera portal.

### CODE REVIEW

File: test/xdg_desktop_portal_test.dart [hunk 1]
```diff
@@ -35,6 +35,29 @@ class MockAccountDialog {
...
 }
 
+class MockCamera {
+  final Map<String, DBusValue> options;
+
+  MockCamera(this.options);
+
+  @override
+  int get hashCode => Object.hash(
+      '',
```
Comment (robert-ancell) [inline test/xdg_desktop_portal_test.dart:45]: This is not required - just return the result of `Object.hashAll`

**Review by robert-ancell — APPROVED** [review]
I'm not sure how useful this is without pipewire support in Dart but no harm in implementing.

It would be nice to also implement the `IsCameraPresent` property as well, though that could be in a separate PR.

File: test/xdg_desktop_portal_test.dart (hunk 1 above)
Comment (aospiridonov) [reply test/xdg_desktop_portal_test.dart]: Done

**Comment by aospiridonov** [comment]
> I'm not sure how useful this is without pipewire support in Dart but no harm in implementing.

Yes, dart is not support pipewire. Can you suggest what needs to be done for this? Or make a repository and try to make it together too. If it's relevant of course.

> It would be nice to also implement the `IsCameraPresent` property as well, though that could be in a separate PR.

I was thinking about property `IsCameraPresent`. Can make some kind of common interface? Since there are properties in other portals.

**Comment by robert-ancell** [comment]
> > I'm not sure how useful this is without pipewire support in Dart but no harm in implementing.
> 
> Yes, dart is not support pipewire. Can you suggest what needs to be done for this? Or make a repository and try to make it together too. If it's relevant of course.
> 

I haven't investigated what we can do with Dart and Pipewire yet, but you're welcome to try!

> > It would be nice to also implement the `IsCameraPresent` property as well, though that could be in a separate PR.
> 
> I was thinking about property `IsCameraPresent`. Can make some kind of common interface? Since there are properties in other portals.

I would just make a `getIsCameraPresent` method. Since portals don't support the org.freedesktop.DBus.Properties interface or have a signal we can't tell if the value changes.

**Comment by aospiridonov** [comment]
> > > I'm not sure how useful this is without pipewire support in Dart but no harm in implementing.
> > 
> > 
> > Yes, dart is not support pipewire. Can you suggest what needs to be done for this? Or make a repository and try to make it together too. If it's relevant of course.
> 
> I haven't investigated what we can do with Dart and Pipewire yet, but you're welcome to try!
Okay, I'll try to investigate.
> 
> > > It would be nice to also implement the `IsCameraPresent` property as well, though that could be in a separate PR.
> > 
> > 
> > I was thinking about property `IsCameraPresent`. Can make some kind of common interface? Since there are properties in other portals.
> 
> I would just make a `getIsCameraPresent` method. Since portals don't support the org.freedesktop.DBus.Properties interface or have a signal we can't tell if the value changes.
Okay, I'll do it. In this PR or in a separate one?

**Comment by robert-ancell** [comment]
> > > > It would be nice to also implement the `IsCameraPresent` property as well, though that could be in a separate PR.
> > > 
> > > 
> > > I was thinking about property `IsCameraPresent`. Can make some kind of common interface? Since there are properties in other portals.
> > 
> > 
> > I would just make a `getIsCameraPresent` method. Since portals don't support the org.freedesktop.DBus.Properties interface or have a signal we can't tell if the value changes.
> 
> Okay, I'll do it. In this PR or in a separate one?

Actually, now I look more closely it does implement `org.freedesktop.DBus.Properties` - so it should be a `Stream<bool>`. Let's do it in a separate PR then.

File: lib/src/xdg_camera_portal.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import 'xdg_portal_request.dart';
+
+/// Camera portal.
+class XdgCameraPortal {
+  final DBusRemoteObject _object;
+  final String Function() _generateToken;
+
+  XdgCameraPortal(this._object, this._generateToken);
+
+  /// Check to access camera.
```
Comment (robert-ancell) [inline lib/src/xdg_camera_portal.dart:14]: I [improved the documentation upstream](https://github.com/flatpak/xdg-desktop-portal/pull/857). A cleared description is "Request to gain access to the camera".

File: lib/src/xdg_camera_portal.dart [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  Future<void> accessCamera() async {
```
Comment (robert-ancell) [inline lib/src/xdg_camera_portal.dart:15]: I think this should be a stream, as the camera access can be closed by the portal later.

**Pull request merged**